modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/PlaylistUpdateAnalytics.kt

```diff
@@ -0,0 +1,236 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsEvent
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTrackerWrapper
+import au.com.shiftyjelly.pocketcasts.models.entity.Playlist
+import au.com.shiftyjelly.pocketcasts.repositories.extensions.colorIndex
+import au.com.shiftyjelly.pocketcasts.repositories.extensions.drawableId
+import timber.log.Timber
+import javax.inject.Inject
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+
+class PlaylistUpdateAnalytics @Inject constructor(
+    private val analyticsTracker: AnalyticsTrackerWrapper
+) {
+
+    fun update(
+        playlist: Playlist,
+        userPlaylistUpdate: UserPlaylistUpdate?,
+        isCreatingFilter: Boolean
+    ) {
+
+        when {
+            isCreatingFilter -> sendPlaylistCreatedEvent(playlist)
+
+            // If the playlist is a draft, then we are in the middle of the filter creation flow and
+            // we don't want to send update events
+            !playlist.draft -> sendPlaylistUpdateEvent(userPlaylistUpdate)
+        }
+    }
+
+    private fun sendPlaylistCreatedEvent(playlist: Playlist) {
+        val properties = buildMap<String, Any> {
+
+            put(Key.ALL_PODCASTS, playlist.allPodcasts)
+            colorAnalyticsValue(playlist)?.let {
+                put(Key.COLOR, it)
+            }
+            put(Key.DOWNLOADED, playlist.downloaded)
```

Comment:
This is missing downloading and not downloaded but it isn't in the spreadsheet so not an issue with this pull request. 

The download option of "all" has downloaded, downloading, and not downloaded set to true. So it might not be that much use without the other flags.

Reply:
Thanks for catching this!

I added a `not_downloaded` boolean prop in [1e311e4](https://github.com/Automattic/pocket-casts-android/pull/414/commits/1e311e46dd22424a656ea306c5dc372f85b1a933).

I'm less clear on what the `downloading` value does though. It looks like the download option of "all" [only sets `downloaded` and `notDownloaded` to `true`](https://github.com/Automattic/pocket-casts-android/blob/1e02c85fe568047f2dc95baf41268e70fed100ed/modules/features/filters/src/main/java/au/com/shiftyjelly/pocketcasts/filters/TimeOptionsFragment.kt#L209-L213).  It does not update `downloading`. In fact, I don't see any way that a user can alter the `downloading` property on a filter. Is there a bug here because the `downloading` property is never getting updated by the client?

Reply:
Oh I got that wrong, we must have taken `downloading` out the UI at some point. I found this comment in the code:

```
// regular filters no longer have a downloading but the not downloaded one should show in progress downloads
```

https://github.com/Automattic/pocket-casts-android/blob/1e02c85fe568047f2dc95baf41268e70fed100ed/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/PlaylistManagerImpl.kt#L367